src/ir/datalog/program.h

```diff
@@ -92,6 +92,11 @@ class ArgumentType {
 
   std::string DebugPrint() const { return absl::StrCat(kind_, name_); }
 
+  bool operator==(const ArgumentType& otherType) const {
+    return this->kind_ == otherType.kind() &&
```

Comment:
Can you add a comment as to why we ignore the name?

Reply:
fixed